pkg/util/kubernetes/kubelet/kubelet.go

```diff
@@ -120,6 +124,7 @@ func GetKubeUtil() (KubeUtilInterface, error) {
 			RetryDelay:    30 * time.Second,
 		})
 	}
+	globalKubeUtilMutex.Unlock()
```

Comment:
is `globalKubeUtil.initRetry.TriggerRetry()` take time?
else maybe you can use a `defer globalKubeUtilMutex.Unlock()` just after line 116

Reply:
The idea was indeed to try to hold the lock for the minimum amount of time.
But it’s true that, in the nominal case, once the retrier has succeeded once, the `globalKubeUtil.initRetry.TriggerRetry()` will return immediately.
So, let’s go for the idiomatic way.
Change done!